core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -38,11 +38,17 @@ trait RTS {
     context.runSync
   }
 
-  final def unsafeShutdownAndWait(timeout: Duration.Finite): Unit = {
+  final def unsafeShutdownAndWait(timeout: Duration): Unit = {
+
+    def awaitTermination(es: ExecutorService) = timeout match {
+      case d: Duration.Finite  => es.awaitTermination(d.toMillis, TimeUnit.MILLISECONDS)
+      case Duration.Infinite() => while (es.isTerminated) {}
```

Comment:
Rather than spin-locking here, do something like this:

```scala
while (!es.isTerminated) es.awaitTermination(Long.MaxValue, TimeUnit.YEARS)
```

Reply:
YEARS is a part of ChronoUnit. TimeUnit.DAYS will have to suffice.